{
  try {
    awaitAndThrowOnError(launderingException);
  }
 catch (  InterruptedException e) {
    throw handleInterrupted(e);
  }
}
